lib/Runtime/DeferredWeightLoader.cpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Glow Contributors. See CONTRIBUTORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Runtime/DeferredWeightLoader.h"
+
+namespace glow {
+namespace runtime {
+
+Error DeferredWeightLoader::setSrc(void *loaderObject) {
+  return MAKE_ERR(ErrorValue::ErrorCode::RUNTIME_ERROR,
+                  "Error weight loader not registered.");
+  ;
+}
+Error DeferredWeightLoader::loadNextWeight() {
+  return MAKE_ERR(ErrorValue::ErrorCode::RUNTIME_ERROR,
+                  "Error weight loader not registered.");
+}
+
+std::string DeferredWeightLoader::getName() { return ""; }
+
+Tensor *DeferredWeightLoader::getTensor() { return nullptr; }
+DeferredWeightLoader *DeferredWeightLoaderRegistry::getLoader() {
+  return loader_;
+}
+
+void DeferredWeightLoaderRegistry::registerLoader(
+    DeferredWeightLoader *loader) {
+  loader_ = loader;
+}
+DeferredWeightLoaderRegistry *DeferredLoader() {
+  static auto *deferredLoader = new DeferredWeightLoaderRegistry();
+  deferredLoader->registerLoader(new DeferredWeightLoader());
```

Comment:
I don't think you want to instantiate a  `DeferredWeightLoader` inside this method, just a `DeferredWeightLoaderRegistry`.  You want to call `registerLoader` from a static instantiation of a subclass of `DeferredWeightLoader`.  See how this is done for StatsExporter: https://github.com/pytorch/glow/blob/master/tests/unittests/StatsExporterTest.cpp

Reply:
Good catch, the testDeferredWeightLoader is registered in the test and used for the test. Removed the instantiation of DeferredWeightLoader here.